lib/src/header_parser/type_extractor/extractor.dart

```diff
@@ -100,17 +99,17 @@ Type getCodeGenType(
       return _extractFromFunctionProto(cxtype);
     case clang_types.CXTypeKind
         .CXType_ConstantArray: // Primarily used for constant array in struct members.
-      return Type.constantArray(
+      return ConstantArray(
         clang.clang_getNumElements(cxtype),
         clang.clang_getArrayElementType(cxtype).toCodeGenType(),
       );
     case clang_types.CXTypeKind
         .CXType_IncompleteArray: // Primarily used for incomplete array in function parameters.
-      return Type.incompleteArray(
+      return IncompleteArray(
```

Comment:
I think we could have some static functions on Type class to create objects of its subtypes. So instead of doing PointerType() we could still do Type.pointer(). This probably would have reduced the diff a lot.

Reply:
Not suggesting a change, but just an observation.

Reply:
Yeah, it would have reduced the diff. But I'm not sure what the point of those functions would be in the new system. It seemed simpler to me to just remove them.